### PR TITLE
duckdb: add version 1.0.0

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **duckdb/1.0.0**

https://github.com/duckdb/duckdb/compare/v0.10.2...v1.0.0

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/duckdb/all/conandata.yml [hunk 1]
```diff
@@ -1,4 +1,7 @@
 sources:
+  "0.10.3":
+    url: "https://github.com/duckdb/duckdb/archive/refs/tags/v0.10.3.tar.gz"
```
Comment (Croydon) [inline recipes/duckdb/all/conandata.yml:3]: They are offering official source downloads, so please use them instead of the auto-generated tarballs

Comment (toge) [reply recipes/duckdb/all/conandata.yml:3]: @Croydon 
Are you referring to libduckdb-src.zip?
It contains only three files (duckdb.cpp, duckdb.h, duckdb.hpp) and duckdb.cpp is too large to compile in the CCI ci environment.

Comment (Croydon) [reply recipes/duckdb/all/conandata.yml:3]: Ah, my bad, sorry

**Review by Croydon — APPROVED** [review]

File: recipes/duckdb/all/patches/1.0.0-0002-msvc-bicobj.patch [hunk 2]
```diff
@@ -0,0 +1,13 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 50d10e8..a06f30c 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -537,7 +537,7 @@ if(NOT MSVC)
+   endif()
+ else()
+   set(CMAKE_CXX_WINDOWS_FLAGS
+-      "/wd4244 /wd4267 /wd4200 /wd26451 /wd26495 /D_CRT_SECURE_NO_WARNINGS /utf-8")
```
Comment (memsharded) [inline recipes/duckdb/all/patches/1.0.0-0002-msvc-bicobj.patch:9]: Isn't it enough with https://github.com/duckdb/duckdb/blob/5e3c44147b2abe0509165fbcd13b926e772679ff/src/CMakeLists.txt#L41

**Review by memsharded — APPROVED** [review]
Looks good thanks very much for your contribution @toge 

There is one minor comment, but as it wouldn't harm, I am approving

It is also possible to remove older patches, 0.10.1, 0.9.0, 0.8.0, 0.6.0 can be safely dropped too (not necessary to do it now, not necessary to do it as a later PR just for this, just as a hint for next time editing this recipe)

**Comment by Taepper** [comment]
Is there anything blocking this PR?

**Comment by memsharded** [comment]
I was waiting on @toge feedback regarding the above comment, but as it isn't a blocker I'll try to move this forward.

**Review by RubenRBS — APPROVED** [review]

**Pull request merged**